src/portfft/common/logging.hpp

```diff
@@ -261,6 +313,71 @@ struct global_data_struct {
   }
 };
 
+/*
+ * Outputs an object to std::cout. Most objects are piped directly to std::cout.
+ *
+ * @tparam T type of the object to output
+ * @param object object to output
+ */
+template <typename T, typename std::enable_if_t<!std::is_enum_v<T>>* = nullptr>
+__attribute__((always_inline)) inline void output(T object) {
```

Comment:
The only thing I can think of would be std::string. Which we currently do not use, but I can add const& just in case. I see no point in perfect forwarding, as operator<< on cout does not have optimized overloads for rvalues.